### PR TITLE
chores: change string type to duration type

### PR DESCRIPTION
**Pull request by dongjiang1989**
## Description

_Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request.
If it fixes a bug or resolves a feature request, be sure to link to that issue._

use built-in type for validating durations type.

## Type of change

_What type of changes does your code introduce to the Prometheus operator? Put an `x` in the box that apply._

- [x] `CHANGE` (fix or feature that would cause existing functionality to not work as expected)
- [ ] `FEATURE` (non-breaking change which adds functionality)
- [ ] `BUGFIX` (non-breaking change which fixes an issue)
- [ ] `ENHANCEMENT` (non-breaking change which improves existing functionality)
- [ ] `NONE` (if none of the other choices apply. Example, tooling, build system, CI, docs, etc.)

## Verification

Please check the [Prometheus-Operator testing guidelines](../TESTING.md) for recommendations about automated tests.

## Changelog entry

_Please put a one-line changelog entry below. This will be copied to the changelog file during the release process._

```release-note
Add validation for duration fields in remote-write configurations.
```

### CODE REVIEW

File: pkg/apis/monitoring/v1/prometheus_types.go [hunk 1]
```diff
@@ -1309,13 +1309,13 @@ type QueueConfig struct {
 	// MaxSamplesPerSend is the maximum number of samples per send.
 	MaxSamplesPerSend int `json:"maxSamplesPerSend,omitempty"`
 	// BatchSendDeadline is the maximum time a sample will wait in buffer.
-	BatchSendDeadline string `json:"batchSendDeadline,omitempty"`
+	BatchSendDeadline *Duration `json:"batchSendDeadline,omitempty"`
```
Comment (simonpasquier) [inline pkg/apis/monitoring/v1/prometheus_types.go:1312]: should be marked as optional
```suggestion
	// +optional
	BatchSendDeadline *Duration `json:"batchSendDeadline,omitempty"`
```

File: pkg/apis/monitoring/v1/prometheus_types.go [hunk 2]
```diff
@@ -1309,13 +1309,13 @@ type QueueConfig struct {
...
 	// MaxRetries is the maximum number of times to retry a batch on recoverable errors.
 	MaxRetries int `json:"maxRetries,omitempty"`
 	// MinBackoff is the initial retry delay. Gets doubled for every retry.
-	MinBackoff string `json:"minBackoff,omitempty"`
+	MinBackoff *Duration `json:"minBackoff,omitempty"`
```
Comment (simonpasquier) [inline pkg/apis/monitoring/v1/prometheus_types.go:1316]: same

File: pkg/apis/monitoring/v1/prometheus_types.go [hunk 3]
```diff
@@ -1309,13 +1309,13 @@ type QueueConfig struct {
...
 	// MaxBackoff is the maximum retry delay.
-	MaxBackoff string `json:"maxBackoff,omitempty"`
+	MaxBackoff *Duration `json:"maxBackoff,omitempty"`
```
Comment (simonpasquier) [inline pkg/apis/monitoring/v1/prometheus_types.go:1318]: same

File: pkg/prometheus/promcfg_test.go [hunk 4]
```diff
@@ -3541,10 +3541,10 @@ func TestRemoteWriteConfig(t *testing.T) {
 					MinShards:         1,
 					MaxShards:         10,
 					MaxSamplesPerSend: 100,
-					BatchSendDeadline: "20s",
+					BatchSendDeadline: (*monitoringv1.Duration)(ptr.To("20s")),
```
Comment (simonpasquier) [inline pkg/prometheus/promcfg_test.go:3544]: (nit)
```suggestion
					BatchSendDeadline: ptr.To(monitoringv1.Duration("20s")),
```

**Review by simonpasquier — COMMENTED** [review]
It's technically a breaking change since it's more restrictive than before. But any existing spec with an invalid duration value is already broken.

File: pkg/prometheus/promcfg_test.go (hunk 4 above)
Comment (dongjiang1989) [reply pkg/prometheus/promcfg_test.go]: changed this nit

File: pkg/apis/monitoring/v1/prometheus_types.go (hunk 2 above)
Comment (dongjiang1989) [reply pkg/apis/monitoring/v1/prometheus_types.go:1318]: Done

**Comment by dongjiang1989** [comment]
> It's technically a breaking change since it's more restrictive than before. But any existing spec with an invalid duration value is already broken.

Thanks for your review. @simonpasquier, all (nits) changed.  Please-recheck

**Review by simonpasquier — APPROVED** [review]
Thanks!

**Pull request merged**